docs/extended_api/synchronization_primitives/atomic_ref.md

```diff
@@ -0,0 +1,101 @@
+---
+grand_parent: Extended API
+parent: Synchronization Primitives
+nav_order: 1
+---
+
+# `cuda::atomic_ref`
+
+Defined in header `<cuda/atomic>`:
+
+```cuda
+template <typename T, cuda::thread_scope Scope = cuda::thread_scope_system>
+class cuda::atomic_ref;
+```
+
+The class template `cuda::atomic_ref` is an extended form of [`cuda::std::atomic_ref`]
+  that takes an additional [`cuda::thread_scope`] argument, defaulted to
+  `cuda::std::thread_scope_system`.
+It has the same interface and semantics as [`cuda::std::atomic_ref`], with the
+  following additional operations.
+This class additionally deviates from the standard by being backported to C++11.
+
+## Limitations
+
+`cuda::atomic_ref<T>` and `cuda::std::atomic_ref<T>` may only be instantiated with types sized between 4 and 8 bytes.
+
+Constructing multiple `cuda::std::atomic_ref<T, Sco>` with the same reference, but different values for `Sco` is illegal.
```

Comment:
I see this as similar to the language spec about forming an `atomic_ref` around a suboject of an object referenced by another `atomic_ref`:

> [No subobject of the object referenced by atomic_­ref shall be concurrently referenced by any other atomic_­ref object](https://eel.is/c++draft/atomics.ref.generic#general-3.sentence-3)

I suspect this is because of the following:

> [Note [1](https://eel.is/c++draft/atomics.ref.generic#general-note-1): Atomic operations or the atomic_­ref constructor can acquire a shared resource, such as a lock associated with the referenced object, to enable atomic operations to be applied to the referenced object[.](https://eel.is/c++draft/atomics.ref.generic#general-4.sentence-2)
— end note]

The constructor of `atomic_ref` could acquire a lock, and therefore constructing multiple `atomic_ref`s with different scopes could deadlock.

Reply:
Oh, I see what you're saying. I didn't read it that way, but I see why you did. I think the change you suggested makes sense.